deps: Fix install on newer ubuntu distros and drop python2 support.

jameshilliard opened the pull request:
Dependencies such as jefferson are now python3 only and python2 is no longer supported, as it appears all our dependencies have python3 versions it should be safe to drop python2 support.

cclauss commented on .travis.yml:
```diff
@@ -1,3 +1,8 @@
+sudo: required
```
Travis CI has deprecated the sudo: tag.

jameshilliard replied:
Removed tag.

cclauss commented on .travis.yml:
```diff
@@ -10,11 +15,16 @@ python:
+  include:
+   - python: "3.6"
+     dist: bionic
+   - python: "3.8"
```
Add Python 3.9?

jameshilliard replied:
Added

cclauss commented on .travis.yml:
```diff
@@ -10,11 +15,16 @@ python:
  - ./setup.py install
 script: ./setup.py test
 cache: pip
 matrix:
```
```suggestion
jobs:
```
Travis CI has replaced `matrix` with `jobs`.

jameshilliard replied:
Changed.

cclauss commented:
[flake8](http://flake8.pycqa.org) testing of https://github.com/ReFirmLabs/binwalk on Python 3.9.0

$ __flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics__
```
./src/binwalk/plugins/tar.py:36:22: F821 undefined name 'xrange'
            for i in xrange(len(s) - 1):
                     ^
./src/binwalk/core/magic.py:628:86: F821 undefined name 'long'
                        if isinstance(line.opvalue, int) or isinstance(line.opvalue, long):
                                                                                     ^
./src/binwalk/core/compat.py:87:16: F821 undefined name 'raw_input'
        return raw_input(prompt)
               ^
./src/scripts/binida.py:1:4: F821 undefined name 'idaapi'
if idaapi.IDA_SDK_VERSION <= 695:
   ^
4     F821 undefined name 'raw_input'
4
```
https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, these tests do _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead, these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python, they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python, a __NameError__ is raised which will halt/crash the script on the user.

ffontaine commented on deps.sh:
```diff
@@ -42,23 +42,25 @@ YUMCMD="yum"
 if [ $distro = "Kali" ]
 then
     APT_CANDIDATES="git build-essential libqt4-opengl mtd-utils gzip bzip2 tar arj lhasa p7zip p7zip-full cabextract util-linux firmware-mod-kit cramfsswap squashfs-tools zlib1g-dev liblzma-dev liblzo2-dev sleuthkit default-jdk lzop cpio"
```
Kali should also be updated: libqt4-opengl is no more available and should also be replaced by qt5base-dev

jameshilliard replied:
Applied change.

cclauss reviewed (changes requested):
Also edit `setup.py` to add `classifiers` and `long_description` to provide more data on https://pypi.org/project/binwalk

jameshilliard commented:
@cclauss I've fixed all the flake errors other than the `idaapi` one as I'm not set up to test a fix with IDA Pro. Also added `long_description` and `long_description_content_type` with the `README.md` used for generating the `long_description`.

cclauss reviewed (approved):
Nice!!

jameshilliard commented:
Added `python_requires=">=3"` and changed the shebang to use `python3` instead of `python`.

cclauss commented on setup.py:
```diff
@@ -318,14 +322,21 @@ def run(self):
     author="Craig Heffner",
     url="https://github.com/ReFirmLabs/%s" % MODULE_NAME,
     requires=[],
+    python_requires=">=3",
```
```suggestion
    python_requires=">=3.6",
```
* https://devguide.python.org/#status-of-python-branches
* https://devguide.python.org/devcycle/#end-of-life-branches

jameshilliard replied:
Since I hadn't tested which version was the actual minimum required I just set it to `>=3`.

cclauss replied:
If the Python core team has dropped support then this repo should too unless someone can express good reasons why the support risk is a worthwhile investment.

jameshilliard replied:
Yeah, not really sure there, there tends to be a lot of legacy tooling floating around in the embedded world that could have implications.

eacmen commented:
Thank you to everyone for the hard work on this. This transition has been a long time coming, apologies it took us so long to address this issue. @jameshilliard @cclauss @ffontaine 👏
